client/client.go

```diff
@@ -113,9 +113,10 @@ type Client struct {
 	servers *serverlist
 
 	// heartbeat related times for tracking how often to heartbeat
-	lastHeartbeat time.Time
-	heartbeatTTL  time.Duration
-	heartbeatLock sync.Mutex
+	lastHeartbeat   time.Time
```

Comment:
Maybe have a heartbeat struct instead of all of these as direct fields on the client?